docs/current/sdk/go/guides/421437-work-with-host-filesystem.md

```diff
@@ -0,0 +1,283 @@
+
+The following example shows how to list the contents of the host working directory:
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+  "log"
+
+  "dagger.io/dagger"
+)
+
+func main() {
+  ctx := context.Background()
+
+  client, err := dagger.Connect(ctx, dagger.WithWorkdir("."))
+  if err != nil {
+    log.Fatal(err)
+  }
+  defer client.Close()
+
+  // highlight-start
+  entries, err := client.Host().Workdir().Entries(ctx)
+  // highlight-end
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  fmt.Println(entries)
+}
+```
+
+## List directory contents with filters
+
+It's possible to restrict a `Directory` to a subset of directory entries, by specifying a list of filename patterns to include or exclude.
+
+The following example shows how to obtain a reference to the host working directory containing only `*.rar` files:
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+  "log"
+  "os"
+  "path/filepath"
+
+  "dagger.io/dagger"
+)
+
+func main() {
+  dir := os.TempDir()
+  os.WriteFile(filepath.Join(dir, "foo.txt"), []byte("1"), 0600)
+  os.WriteFile(filepath.Join(dir, "bar.txt"), []byte("2"), 0600)
+  os.WriteFile(filepath.Join(dir, "baz.rar"), []byte("3"), 0600)
+
+  ctx := context.Background()
+
+  client, err := dagger.Connect(ctx, dagger.WithWorkdir(dir))
+  if err != nil {
+    log.Fatal(err)
+  }
+  defer client.Close()
+
+  // highlight-start
+  entries, err := client.Host().Workdir(dagger.HostWorkdirOpts{
+    Include: []string{"*.rar"},
+  }).Entries(ctx)
+  // highlight-end
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  fmt.Println(entries)
+}
+```
+
+The following example shows how to obtain a reference to the host working directory containing all files except `*.txt` files:
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+  "log"
+  "os"
+  "path/filepath"
+
+  "dagger.io/dagger"
+)
+
+func main() {
+  dir := os.TempDir()
+  os.WriteFile(filepath.Join(dir, "foo.txt"), []byte("1"), 0600)
+  os.WriteFile(filepath.Join(dir, "bar.txt"), []byte("2"), 0600)
+  os.WriteFile(filepath.Join(dir, "baz.rar"), []byte("3"), 0600)
+
+  ctx := context.Background()
+
+  client, err := dagger.Connect(ctx, dagger.WithWorkdir(dir))
+  if err != nil {
+    log.Fatal(err)
+  }
+  defer client.Close()
+
+  // highlight-start
+  entries, err := client.Host().Workdir(dagger.HostWorkdirOpts{
+    Exclude: []string{"*.txt"},
+  }).Entries(ctx)
+  // highlight-end
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  fmt.Println(entries)
+}
+```
+
+:::note
+The `Exclude` pattern overrides the `Include` pattern, but not vice-versa.
+:::
+
+## Mount a host directory in a container
+
+A common operation when working with containers is to mount a host directory to a path in the container and then perform operations on it. This can be done using the `Container.WithMountedDirectory()` function, which accepts the mount point in the container and the `Directory` to be mounted as arguments.
```

Comment:
While I'm splitting hairs...

```suggestion
A common operation when working with containers is to mount a host directory to a path in the container and then perform operations on it. This can be done using the `Container.WithMountedDirectory()` method, which accepts the mount point in the container and the `Directory` to be mounted as arguments.
```